doc/docs/master/how-tos/developer-workflow/build-pipelines.md

```diff
@@ -0,0 +1,118 @@
+
+Using this feature requires a modification of the pipeline spec's `transform` object with options for the following fields:
+
+- `path`: An optional string specifying where the source code is relative to the pipeline spec path (or the cwd if the pipeline is fed into `pachctl` via stdin.)
+- `language`: An optional string specifying what language builder to use (see below.) Only works with official builders. If unspecified, `image` will be used instead.
+- `image`: An optional string specifying what builder image to use, if a non-official builder is desired. If unspecified, the `transform` object's `image` will be used instead.
+
+Below is a Python example of a build pipline.
+
+```json
+{
+  "pipeline": {
+    "name": "map"
+  },
+  "description": "A pipeline that tokenizes scraped pages and appends counts of words to corresponding files.",
+  "transform": {
+    "build": {
+      "language": "python",
+      "path": "./source"
+    }
+  },
+  "input": {
+    "pfs": {
+      "repo": "scraper",
+      "glob": "/*"
+    }
+  }
+}
+```
+
+A build pipeline can be submitted the same way as any other pipeline, for example:
+
+```bash
+pachctl update pipeline -f <pipeline name>
+```
+
+## How it works
+
+When submitted, the following actions occur:
+
+1. All files (code, etc.) are copied from the build path to a PFS repository, `<pipeline name>_build`, which we can think of as the source code repository. In the case above, everything in `./source` would be copied to to the PFS `<pipeline name>_build` repository.
+
+1. A pipeline called `<pipeline name>_build` starts, reading reading files from the `<pipeline name>_build` repo and runs the `build.sh` script to pull dependencies and compile any requirements from the source code.
+
+1. The running pipeline, `<pipeline name>`, is updated (or created if it doesn't already exist) with the the new source files and built assets.
+
+!!! note
+      You can optionally specify a `.pachignore` file, which uses [ohmyglob](https://github.com/pachyderm/ohmyglob) entries to prevent certain files from getting pushed to this repo.
+
+The updated pipeline contains the following PFS repos mapped in as inputs:
+
+1. `/pfs/source` - source code that is required for running the pipeline. There are language-specific requirements on the structure of this source code mentioned in the [Python Builder](#python-builder) section.
+
+1. `/pfs/build` - any artifacts resulting from the build process.
+
+1. `/pfs/<input(s)>` - any inputs specified in the pipeline specification.
+
+## Builders
+The builder is responsible for determining what Docker image is used for the base image and what steps to run during the pipeline build and deployment. If a builder has a standard implementation (currently `python` and `go`), then only the `transform.build.language` needs to be set, as it has impelmentations for `build.sh` and `run.sh`. The order of preference is:
```

Comment:
I'm not sure how to rephrase this, but it's a little confusing atm, because the builder itself isn't responsible for that -- rather, the pipeline spec is.

Reply:
Good catch, are you okay with: 
"The builder interprets the pipeline spec to determine what Docker image to use as the base image and what steps to run for the build and deployment."